crates/module-system/module-implementations/sov-evm/src/authenticate/eip712.rs

```diff
@@ -89,14 +134,41 @@ where
         capabilities::AuthenticationOutput<S, Self::Decodable>,
         capabilities::UnregisteredAuthenticationError,
     > {
-        let tx: RawTx = borsh::from_slice(&batch.tx.data)
-            .map_err(|_| UnregisteredAuthenticationError::InvalidAuthenticationDiscriminant)?;
+        let Self::Input::Standard(input) = borsh::from_slice(&batch.tx.data)
+            .map_err(|_| UnregisteredAuthenticationError::InvalidAuthenticationDiscriminant)?
+        else {
+            return Err(UnregisteredAuthenticationError::InvalidAuthenticationDiscriminant);
+        };
+
+        let (tx_and_raw_hash, auth_data, runtime_call) =
+            sov_modules_api::capabilities::authenticate::<_, S, Rt>(
+                &input.data,
+                &Rt::CHAIN_HASH,
+                state,
+            )
+            .map_err(|e| match e {
+                AuthenticationError::FatalError(err, hash) => {
```

Comment:
Can we implement `Into<UnregisteredAuthenticationError> for AuthenticationError` and do just `.map_err(Into::into) here`?

Reply:
Good suggestion - though I think unregistered authentication should be refactored a bit (the entire function is more or less duplicated in every authenticator) so it's probably best done in a separate PR